gateway/src/http_entrypoints.rs

```diff
@@ -172,6 +171,26 @@ pub enum SpType {
     Switch,
 }
 
+impl From<SpType> for gateway_sp_comms::SpType {
+    fn from(typ: SpType) -> Self {
+        match typ {
+            SpType::Sled => Self::Sled,
+            SpType::Power => Self::Power,
+            SpType::Switch => Self::Switch,
+        }
+    }
+}
+
+impl From<gateway_sp_comms::SpType> for SpType {
```

Comment:
These two `From` impls (and similarly below for `SpIdentifer`) are a bit of an eyesore; open to suggestions. Seems a little weird to have them derive `JsonSchema` etc from inside `gateway_sp_comms`, but maybe that's the simplest thing?

Reply:
I guess we could have it derive `JsonSchema` from `gateway_sp_comms` under some feature flag so that the Hubris task isn't burdened with it. I don't love it for the same reasons as I think you don't... on the other hand, it seems not-uncommon to have a feature to impl stuff like `Serialize`/`Deserialize` so maybe this amounts to the same thing